src/redownload/downloads.py

```diff
@@ -0,0 +1,28 @@
+import os
+
+import requests
+
+from . import exceptions
+
+
+def download_from_set(urls: set, out_dir: str) -> None:
+    """Takes a set of urls as an arg and downloads them all to out_dir.
+
+    :param urls: a set of URLs.
+    :param out_dir: a string containing a path pointing to the directory to save the downloaded files to
+    :return: None
+    """
+    # If the output dir is a file, raise an exception.
+    if os.path.isfile(out_dir):
+        raise exceptions.InvalidOutputDir(
+            "The output path specified is a file and cannot be overwritten."
+        )
+    # If the output dir doesn't exist, create it.
+    if not os.path.exists(out_dir):
+        os.mkdir(out_dir)
+    # Download all the URLs in the set to the output dir.
+    for url in urls:
+        filename = url[url.rfind("/") + 1 : len(url)]
+        request = requests.get(url)
+        with open(os.path.join(out_dir, filename), "wb") as file:
+            file.write(request.content)
```

Comment:
Might be better to write the file in chunks.

```python
    try:
        response = requests.get(url, stream=True)
    except requests.RequestException as e:
        print(e)

    with open(os.path.join(out_dir, filename), "wb") as file:
        for chunk in response.iter_content(chunk_size=1024):
            if chunk:
                file.write(chunk)
```

Reply:
I'll look into the second part there. 

I'm intentionally not catching exceptions on the read because I want it to fail there if theres something wrong with the request (nothing else in the function will work if that fails).

Reply:
What's the purpose of `if chunk:` there?

Reply:
Unless you require the traceback, maybe something like this.

```python
print(e)  # or some meaningful message
sys.exit(1)
```

Reply:
or just let it raise the exception and exit on it's own?